tools/run_tests/xds_k8s_test_driver/bin/cleanup/cleanup.py

```diff
@@ -0,0 +1,374 @@
+        td_bootstrap_image=xds_k8s_flags.TD_BOOTSTRAP_IMAGE.value,
+        gcp_project=project,
+        gcp_api_manager=gcp.api.GcpApiManager(),
+        gcp_service_account=gcp_service_account,
+        network=network)
+
+    server_namespace = KubernetesServerRunner.make_namespace_name(
+        resource_prefix, resource_suffix)
+    server_runner = KubernetesServerRunner(
+        k8s.KubernetesNamespace(k8s_api_manager, server_namespace),
+        **runner_kwargs)
+
+    logger.info('Cleanup server')
+    server_runner.cleanup(
+        force=True, force_namespace=True
+    )
+
+
+def main(argv):
+    if len(argv) > 1:
+        raise app.UsageError('Too many command-line arguments.')
+    load_keep_config()
+
+    project: str = xds_flags.PROJECT.value
+    network: str = xds_flags.NETWORK.value
+    gcp_service_account: str = xds_k8s_flags.GCP_SERVICE_ACCOUNT.value
+    dry_run: bool = DRY_RUN.value
+
+    # List resources older than KEEP_PERIOD. We only list health-checks and
+    # instance templates because these are leaves in the resource dependency tree.
+    #
+    # E.g. forwarding-rule depends on the target-proxy. So leaked
+    # forwarding-rule indicates there's a leaked target-proxy (because this
+    # target proxy cannot deleted unless the forwarding rule is deleted). The
+    # leaked target-proxy is guaranteed to be a super set of leaked
+    # forwarding-rule.
+    leakedHealthChecks = exec_gcloud(project, 'compute', 'health-checks',
+                                     'list')
+    for resource in leakedHealthChecks:
+        logger.info('-----')
+        logger.info('----- Cleaning up health check %s', resource['name'])
+        if dry_run:
+            # Skip deletion for dry-runs
+            logging.info('----- Skipped [Dry Run]: %s', resource['name'])
+            continue
+
+        # Cleanup resources from the gce framewok.
+        result = re.search(r'test-hc(.*)', resource['name'])
+        if result is not None:
+            if is_marked_as_keep_gce(result.group(1)):
+                logging.info(
+                    'Skipped [keep]: GCE resource suffix [%s] is marked as keep',
+                    result.group(1))
+                continue
+            remove_relative_resources_run_xds_tests(project, result.group(1))
+            continue
+
+        # Cleanup resources from the gke framework.
+        result = re.search(f'{PSM_SECURITY_PREFIX}-health-check-(.*)',
+                           resource['name'])
+        if result is not None:
+            if is_marked_as_keep_gke(result.group(1)):
+                logging.info(
+                    'Skipped [keep]: GKE resource suffix [%s] is marked as keep',
+                    result.group(1))
+                continue
+            cleanup_td_for_gke(project, network, PSM_SECURITY_PREFIX,
+                               result.group(1))
+            continue
+
+        logging.info(
+            '----- Skipped [does not matching resource name templates]')
```

Comment:
We could potentially make health-check handling and the instance-template handling into a function to reduce LOC.

Reply:
There are some subtle differences:
- health-check is for both gce and gke, but instance-template is only for gce